sdk/servicebus/azure-servicebus/azure/servicebus/_common/auto_lock_renewer.py

```diff
@@ -91,23 +96,57 @@ def __init__(
         :type max_workers: Optional[int]
         """
         self._executor = executor or ThreadPoolExecutor(max_workers=max_workers)
+        # None indicates it's unknown whether the provided executor has max worker > 1
+        self._is_max_workers_greater_than_one = None if executor else (max_workers is None or max_workers > 1)
         self._shutdown = threading.Event()
-        self._sleep_time = 1
+        self._sleep_time = 0.5
         self._renew_period = 10
+        self._running = threading.Event()  # indicate whether the main worker thread is running
+        self._last_activity_timestamp = None  # the last timestamp when the main worker is active dealing with tasks
+        self._idle_timeout = 10  # the idle time that main worker thead should exist if there's no activity
         self._max_lock_renewal_duration = max_lock_renewal_duration
         self._on_lock_renew_failure = on_lock_renew_failure
+        self._renew_tasks = queue.Queue()  # type: ignore
+        self._infer_max_workers_time = 1
+        self._infer_max_workers_flags = [0, 0]
 
     def __enter__(self):
         if self._shutdown.is_set():
             raise ServiceBusError(
                 "The AutoLockRenewer has already been shutdown. Please create a new instance for"
                 " auto lock renewing."
             )
+
+        self._init_workers()
```

Comment:
Will it be a problem if the user code isn't using a context manager?

Reply:
the _init_workers is also called in the register method, making sure the non-context manager case is covered